src/google/protobuf/compiler/objectivec/objectivec_file.cc

```diff
@@ -310,6 +310,12 @@ void FileGenerator::GenerateSource(io::Printer *printer) {
   vector<const FileDescriptor*> deps_with_extensions;
   CollectMinimalFileDepsContainingExtensions(file_, &deps_with_extensions);
 
+  printer->Print(
+    "#if __has_feature(objc_arc)\n"
+    " #error This file should not be compiled with ARC. Use the -fobjc-arc Compiler Flag to disable ARC for this file.\n"
```

Comment:
Still wrong, you want -fno-objc-arc.

https://github.com/google/protobuf/tree/master/objectivec#building